src/features/zaaksysteem/zaakContactmomentKoppelaar.vue

```diff
@@ -0,0 +1,46 @@
+<template>
+  <button
+    @click="koppel"
+    class="icon-button icon-after icon-large check"
+    :class="{ selected: selected }"
+  ></button>
+</template>
+<script lang="ts" setup>
+import { useContactmomentStore } from "@/stores/contactmoment";
+import type { Zaak } from "@/features/zaaksysteem/types";
+import { ref, onMounted, type PropType } from "vue";
+
+const props = defineProps({
+  zaak: { type: Object as PropType<Zaak>, default: null },
+});
+
+const contactmoment = useContactmomentStore();
+const selected = ref(false);
+
+const koppel = () => {
+  selected.value = contactmoment.toggleZaak(props.zaak);
+};
+
+onMounted(() => {
+  const stored = contactmoment.findById(props.zaak.id);
```

Comment:
de logica omtrent het wel of niet gekoppeld zijn van een zaak is nu half encapsulated in de store, maar lekt hier naar buiten: de stappen van eerst zoeken op id en als er niks gevonden wordt, dit behandelen als niet gekoppeld, worden nu gedupliceerd. ik heb ook niet echt een panklare oplossing maar de logica wordt nu wel erg verspreid. misschien een getter op de store?

Reply:
deze logica opnemen in de store